larky/src/main/java/com/verygood/security/larky/modules/vgs/cerebro/text/analyzer/TextAnalyzerModule.java

```diff
@@ -0,0 +1,141 @@
+package com.verygood.security.larky.modules.vgs.cerebro.text.analyzer;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Sets;
+import com.verygood.security.larky.modules.vgs.cerebro.text.analyzer.analyzers.DefaultTextPIIAnalyzer;
+import com.verygood.security.larky.modules.vgs.cerebro.text.analyzer.analyzers.NoopTextPIIAnalyzer;
+import com.verygood.security.larky.modules.vgs.cerebro.text.analyzer.dto.TextPIIEntity;
+import com.verygood.security.larky.modules.vgs.cerebro.text.analyzer.spi.TextPIIAnalyzer;
+import net.starlark.java.annot.Param;
+import net.starlark.java.annot.ParamType;
+import net.starlark.java.annot.StarlarkBuiltin;
+import net.starlark.java.annot.StarlarkMethod;
+import net.starlark.java.eval.EvalException;
+import net.starlark.java.eval.Starlark;
+import net.starlark.java.eval.StarlarkFloat;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.ServiceLoader;
+import java.util.Set;
+
+@StarlarkBuiltin(
+    name = "TextPIIAnalyzer",
+    category = "BUILTIN",
+    doc = "Overridable Text PII Analysis API in Larky")
+public class TextAnalyzerModule implements TextPIIAnalyzer {
+
+  public static final TextAnalyzerModule INSTANCE = new TextAnalyzerModule();
+  public static final String ENABLE_INMEMORY_PROPERTY = "larky.modules.vgs.cerebro.piiAnalyzer.text.enableInMemory";
+
+  private TextPIIAnalyzer textPiiAnalyzer;
+
+  public TextAnalyzerModule() {
+
+    ServiceLoader<TextPIIAnalyzer> loader = ServiceLoader.load(TextPIIAnalyzer.class);
+    List<TextPIIAnalyzer> textAnalyzerProviders = ImmutableList.copyOf(loader.iterator());
+
+    if (Boolean.getBoolean(ENABLE_INMEMORY_PROPERTY)) {
```

Comment:
Is this inMemory or is this an overridden module? What would happen if I used this boolean and I also had a service loader pointing to my module. Wouldn't we want some message to tell that there is a clash?

Reply:
Nice catch, added WARN log.